### DISCUSSION TITLE
PushItemsAsync() - SendItemWasPushedEvent sending ItemId as TableName

### DISCUSSION THREAD

**Discussion by acidindustries** [Azure/azure-mobile-apps#721] [Q&A] [answered]
I am trying to use the DataSync library in a custom implementation with LiteDB and MongoDB. The SendItemWasPushedEvent in PushItemsAsync() in SyncContext.cs is sending the tableName when ParallelOperations is set to 1 but it is sending the ItemId as a TableName when ParallelOperations != 1. How come?

Would it be possible to send the intended tableName value in all cases?

Setting ParallelOperations in PushOptions gets around the issue but it is truly a workaround for now.

`csharp
public async Task PushItemsAsync(string[] tableNames, PushOptions options, CancellationToken cancellationToken = default)
        {
            await EnsureContextIsInitializedAsync(cancellationToken).ConfigureAwait(false);

            var batch = new OperationBatch(this);
            cancellationToken.Register(() => batch.Abort(PushStatus.CancelledByToken));

            // Process the queue - only use the QueueHandler (new logic) if maxThreads > 1
            SendPushStartedEvent();
            var maxThreads = GetMaximumParallelOperations(options);
            if (maxThreads == 1)
            {
                try
                {
                    long itemCount = 0;
                    TableOperation operation = await OperationsQueue.PeekAsync(0, tableNames, cancellationToken).ConfigureAwait(false);
                    while (operation != null)
                    {
                        SendItemWillBePushedEvent(operation.TableName, operation.ItemId, itemCount);
                        bool isSuccessful = await ExecutePushOperationAsync(operation, batch, true, cancellationToken).ConfigureAwait(false);
                        itemCount++;
                        SendItemWasPushedEvent(operation.TableName, operation.ItemId, itemCount, isSuccessful);

             ...
             }
            else
            {
                var itemCount = new long[] { 0 };
                QueueHandler queueHandler = new(maxThreads, async (operation) =>
                {
                    SendItemWillBePushedEvent(operation.TableName, operation.ItemId, itemCount[0]);
                    bool isSuccessful = await ExecutePushOperationAsync(operation, batch, true, cancellationToken).ConfigureAwait(false);
                    lock(itemCount)
                    {
                        itemCount[0]++;
                    }
                    ----> SendItemWasPushedEvent(operation.ItemId, operation.ItemId, itemCount[0], isSuccessful);

           ...
         }
`

**Comment by adrianhall** [comment, marked as answer]
Feels like a bug to me.   I'm going to transition this to an issue and will investigate.

**Reply by adrianhall** [reply]
... and the bug was fixed - I'll be sending out a release tonight!

**Comment by acidindustries** [comment]
Perfect! Thanks for the prompt response.

**Reply by adrianhall** [reply]
Released as v5.1.8 - I’ve just approved, and it normally shows up in the Nuget feeds within an hour or so.
